lib/IDE/ConformingMethodList.cpp

```diff
@@ -87,6 +87,10 @@ void ConformingMethodListCallbacks::doneParsing() {
   if (!T || T->is<ErrorType>() || T->is<UnresolvedType>())
     return;
 
+  T = T->getRValueType();
+  if (T->hasArchetype())
+    T = T->mapTypeOutOfContext();
```

Comment:
If the hasArchetype() check is an optimization, you don’t really need it because mapTypeOutOfContext() should already short circuit in this case. If it’s because you might have archetypes or type parameters here, it’s worth untangling that. I would try making it unconditional and seeing if the tests pass.

Reply:
I think it's safe to check this, just in case it have `hasTypeParameter()`. (There's is an invariant that `hasArchetype()` and `hasTypeParameter()` are mutually exclusive, right?)

Reply:
Nothing enforces they're mututally exclusive, that's the problem. This is why I like to establish the invariant by using mapTypeIntoContext()/OutOfContext() unconditionally where possible, and fixing callers to only pass one or the other consistently.

Reply:
I understand. Let me try that in followups.